tests/iothreads_auto_test.go

```diff
@@ -0,0 +1,334 @@
+				Disk: &v1.DiskTarget{
+				    Bus: "virtio",
+				},
+			    },
+			},
+		    },
+		},
+	    },
+	    Volumes: []v1.Volume{
+		{
+		    Name: "ded1volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "ded2volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "shr1volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "shr2volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "shr3volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "shr4volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+	    },
+	}
+
+	var IOThreadVMI *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		IOThreadVMI = tests.NewRandomVMI()
+		IOThreadVMI.Spec = IOThreadSpec
+	})
+
+	Context("IOThreads Policies - auto x2", func() {
+
+		It("Virtual Disk Settings - IOThreads", func() {
+
+			// How many symbols in the names of pod and VMI shoul match
+			symbols_to_compare := 30
+
+			By("Creating VMI with desired spec")
+			IOThreadVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(IOThreadVMI)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(IOThreadVMI.Name) > symbols_to_compare).To(BeTrue(), "VMI Name should contain at least N symbols")
+
+			By("Checking that corresponding pod exists")
+			listOptions := metav1.ListOptions{}
+			podList, err := virtClient.CoreV1().Pods(tests.NamespaceTestDefault).List(listOptions)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(podList.Items).To(HaveLen(1), "We should only have 1 pod")
+			// I don't know how exactly VM and names are formed, and as a result I can't properly compare 
+			// VM names with pod names, since they both have a number or random symbols at the end,
+			// and these symbols are different for pod name and vm name. So, I only can compare their names before
+			// these random symbols.
+			Expect(podList.Items[0].Name).To(HavePrefix("virt-launcher-" + IOThreadVMI.Name[:symbols_to_compare]), "Pod name should have a name similiar to VM name")
+
+			By("Checking that VMI with this name does exist")
+			getOptions := metav1.GetOptions{}
+			resultVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(IOThreadVMI.Name, &getOptions)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(*resultVMI.Spec.Domain.IOThreadsPolicy).To(Equal(autoPolicy), "Spec should have auto policy")
```

Comment:
The create of VM with RandomVMI return VMI object.
on and with this object you can get all you need.
Like: VMI name:   vmi.name
You can run it with: ktests.WaitForSuccessfulVMIStart(vmi) 
And for the connect to contain you have the launcher pod  
pod that runs the VMI: 				
vmiPod := ktests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)

Please fix your code to use it.

Reply:
I've added these function, please check the updated PR. I removed delay that was there to wait for VMI to start (now WaitForSuccessfulVMIStart handles it), but I still receive list of PODs and check that only one POD is exist. If you think it is better to remove this check completely, and only check the name of the POD associated with current VMI - I can remove obtaining list of the pods.